specifications/sep-payment-channel-mechanism.md

```diff
@@ -0,0 +1,657 @@
+close, and that disburse funds to R.
+
+## Specification
+
+### Participants
+
+A payment channel has two participants:
+
+- I, the _initiator_, who proposes the payment channel, performs the first
+setup step, and will be able to make deposits to the payment channel without
+coordination.  I creates escrow account E and receives disbursement through
+regaining control of E at channel close.
+
+- R, the _responder_, who joins the payment channel, and receives disbursement
+through claimable balances at channel close.
+
+### Observation Period
+
+A payment channel defines an observation period O within which all
+participants are guaranteed to be able to observe the blockchain state and
+submit transactions in response to changing state.
+
+The participants agree on the period O at channel open.
+
+The observation period is defined both as a duration in time, and a count of
+ledgers. The observation period has passed if both the duration and ledger count
+have been exceeded. These two properties together are referred to as O
+throughout the protocol.
+
+The participants may agree to change the period O at anytime by following the
+[Change the Observation Period](#Change-the-Observation-Period) process.
+
+### Accounts
+
+The payment channel utilizes two Stellar accounts that are both 2-of-2
+multisig accounts while the channel is open:
+
+- E, the _escrow account_, that holds the assets that both participants have
+contributed to the channel and that will be distributed to the participants
+at channel close according to the final close transactions submitted.
+Created by I.  Jointly controlled by I and R while the channel is open.
+Control is returned to I at close.
+
+- V, the _reserve account_, that holds an amount of native asset contributed
+by R that will be used to sponsor the claimable balance ledger entries
+created at disbursement.  Created by R.  Jointly controlled by I and R while
+the channel is open.  Control is returned to R at close.  Cannot be merged
+until all claimable balances created at close for R are claimed by R.
+
+### Constants
+
+The two participants agree on the following constants:
+
+- m, the _maximum transaction count for an iteration's transaction set_, is
+defined as 2, the maximum number of transactions that can be signed in any
+process between the increments of iteration number i.
+
+### Variables
+
+The two participants maintain the following variables during the lifetime of
+the channel:
+
+- s, the _starting sequence number_, is initialized to one greater than the
+sequence number of escrow account E after E has been created. It is the first
+available sequence number for iterations to consume.
+
+- i, the _iteration number_, is initialized to zero.  It is incremented with
+every off-chain update of the payment channel state, or on-chain setup, deposit,
+withdrawal, etc.
+
+- e, the _executed iteration number_, is initialized to zero. It is incremented
```

Comment:
❓ I'm still not wrapping my head around why we need a formal variable e. It only increments when the payment channel is closed, at which point it doesn't matter

And therefore s_e is only either s, or the sequence # that E ends up at at the close

Reply:
e is set to the i of any transaction that the parties agree to submit to the network, so it is used in all the cases where the parties agree to modify the channel after setup, such as withdrawing.

Reply:
got it. I'm wondering if we can modify e's definition then, I think that's where my confusion stemmed from. What about

```
e, the executed iteration number, is initially set to zero, and thereafter represents the i of any transaction set the parties agree to submit to the network.
```

because it changes when the agreement happens. Not after an iteration is submitted on-chain (eg. in the deposit by R steps it changes before anything is submitted)

Reply:
also, if I'm understanding this right I believe P_i should be bumping E.seqNum to s_e, not s_i. [In which case this line](https://github.com/stellar/experimental-payment-channels/pull/3/files#diff-b01917f0debe4bb9e19259c6e34888dd6c41d4f2da28f49af7fc820ba5a38dc7R478) would be 
```
One BUMP_SEQUENCE operation bumping the sequence number of escrow account E to s_e.
```

Reply:
Ah good point, the description doesn't really say what was intended. Updated in a0b735f. Thank you!

Reply:
> also, if I'm understanding this right I believe P_i should be bumping E.seqNum to s_e, not s_i. [In which case this line](https://github.com/stellar/experimental-payment-channels/pull/3/files#diff-b01917f0debe4bb9e19259c6e34888dd6c41d4f2da28f49af7fc820ba5a38dc7R478) (and the other changing channel setup cases) would be

I think they're the same thing essentially. In the deposit flow in step 2 when P_i is built s_i will be the sequence number of the escrow account we want to bump to. s_e is not currently set to the value we want to bump to, but it will be after in step 4.

Reply:
ah good point. When I read the construction of P_i it was after going through the steps. At the point P_i is created in your steps s_i is right 👍